scrapy/utils/request.py

```diff
@@ -258,32 +141,22 @@ def __init__(self, crawler: Optional["Crawler"] = None):
                 "REQUEST_FINGERPRINTER_IMPLEMENTATION"
             )
         else:
-            implementation = "2.6"
-        if implementation == "2.6":
+            implementation = "SENTINEL"
+
+        if implementation == "SENTINEL":
+            self._fingerprint = fingerprint
+        elif implementation == "2.7":
             message = (
-                "'2.6' is a deprecated value for the "
-                "'REQUEST_FINGERPRINTER_IMPLEMENTATION' setting.\n"
-                "\n"
-                "It is also the default value. In other words, it is normal "
-                "to get this warning if you have not defined a value for the "
-                "'REQUEST_FINGERPRINTER_IMPLEMENTATION' setting. This is so "
-                "for backward compatibility reasons, but it will change in a "
-                "future version of Scrapy.\n"
-                "\n"
-                "See the documentation of the "
-                "'REQUEST_FINGERPRINTER_IMPLEMENTATION' setting for "
-                "information on how to handle this deprecation."
+                "'REQUEST_FINGERPRINTER_IMPLEMENTATION' is a deprecated setting.\n"
+                "And it will be removed in future version of Scrapy."
             )
             warnings.warn(message, category=ScrapyDeprecationWarning, stacklevel=2)
-            self._fingerprint = _request_fingerprint_as_bytes
-        elif implementation == "2.7":
             self._fingerprint = fingerprint
         else:
             raise ValueError(
                 f"Got an invalid value on setting "
                 f"'REQUEST_FINGERPRINTER_IMPLEMENTATION': "
-                f"{implementation!r}. Valid values are '2.6' (deprecated) "
-                f"and '2.7'."
+                f"{implementation!r}. Valid values are '2.7' and 'SENTINEL'."
             )
```

Comment:
What about:

```python
if implementation != "SENTINEL":
    warn(…)
self._fingerprint = fingerprint
```

to replace the whole thing?

i.e. We no longer care if they use an invalid value or not, we ignore whatever value they set, and just warn them to stop using the setting.

Reply:
Done.